kyo-core/shared/src/main/scala/kyo/layers.scala

```diff
@@ -0,0 +1,96 @@
+package kyo
+
+import Layers.internal.*
+import java.util.concurrent.ConcurrentHashMap as JConcurrentHashMap
+import kyo.core.*
+import scala.annotation.targetName
+
+sealed trait Layer[+Out, -S]:
```

Comment:
Kyo's codebase favors `abstract class` over `trait` where possible to avoid interface dispatches, which have some additional performance overhead. Can you check if this could be `abstract class`? Also, please mark the methods as `final` so the JIT can more easily optimize invocations.

Reply:
Done!